addons/mail/static/tests/new/message/message_tests.js

```diff
@@ -1050,3 +1050,43 @@ QUnit.test("allow attachment delete on authored message", async function (assert
     await click(".modal-footer .btn-primary");
     assert.containsNone(target, ".o-mail-attachment-card");
 });
+
+QUnit.test(
+    "prevent attachment delete on non-authored message in channels",
+    async function (assert) {
+        const pyEnv = await startServer();
+        const partnerId = pyEnv["res.partner"].create({});
+        const mailChannelId = pyEnv["mail.channel"].create({ name: "test" });
+        pyEnv["mail.message"].create({
+            attachment_ids: [
+                [
+                    0,
+                    0,
+                    {
+                        mimetype: "image/jpeg",
+                        name: "BLAH",
+                        res_id: mailChannelId,
+                        res_model: "mail.channel",
+                    },
+                ],
+            ],
+            author_id: partnerId,
+            body: "<p>Test</p>",
+            model: "mail.channel",
+            res_id: mailChannelId,
+        });
+        const { openDiscuss } = await start({
+            discuss: {
+                context: { active_id: mailChannelId },
+            },
+        });
+        await openDiscuss();
+
+        assert.containsOnce(target, ".o-mail-attachment-image", "should have an attachment");
+        assert.containsNone(
+            target,
+            ".o-mail-attachment-unlink",
+            "delete attachment button should not be printed"
+        );
```

Comment:
Let's remove assertion comments. They are not that useful, and makes tests wider than they should be.